spec/integration/epoch_interruption_spec.rb

```diff
@@ -0,0 +1,83 @@
+module Wasmtime
+  RSpec.describe "Epoch interruption" do
+    let(:engine) do
+      config = Config.new
+      config.epoch_interruption = true
+      Engine.new(config)
+    end
+
+    let(:store_deadline_0) { Store.new(engine) }
+    let(:store_deadline_1) { Store.new(engine).tap { |store| store.set_epoch_deadline(1) } }
+
+    let(:mod) do
+      Module.new(engine, <<~WAT)
+        (module
+          (func (export "42") (result i32)
+            (i32.const 42))
+          (func (export "loop_forever")
+            (loop br 0)))
+      WAT
+    end
+
+    let(:autostart_mod) do
+      Module.new(engine, <<~WAT)
+        (module
+          (func nop)
+          (start 0))
+      WAT
+    end
+
+    it "starts with epoch deadline 0 and traps immediately" do
+      instance = Instance.new(store_deadline_0, mod)
+
+      expect { instance.invoke("42") }.to raise_error(Trap) do |trap|
+        expect(trap.code).to eq(:interrupt)
+      end
+
+      expect { Instance.new(store_deadline_0, autostart_mod) }.to raise_error(Trap)
+    end
+
+    it "runs to completion when epoch deadline is non-zero" do
+      instance = Instance.new(store_deadline_1, mod)
+      expect(instance.invoke("42")).to eq(42)
+
+      expect { Instance.new(store_deadline_1, autostart_mod) }.not_to raise_error
+    end
+
+    it "allows incrementing epoch manually" do
+      instance = Instance.new(store_deadline_1, mod)
+      # No error: engine is still on epoch 0
+      instance.invoke("42")
+
+      engine.increment_epoch
+      expect { instance.invoke("42") }.to raise_error(Trap)
+    end
+
+    describe "Engine timer" do
+      it "prevents infinite loop from running forever" do
+        instance = Instance.new(store_deadline_1, mod)
+        engine.start_epoch_timer(10)
+        expect { instance.invoke("loop_forever") }.to raise_error(Trap)
```

Comment:
I initially had an assertion here on elapsed time, but it was quite inaccurate in CI, even though it's quite accurate on my M1 mac and on my Linux VM. I've seen [it take less the expected time on Windows](https://github.com/bytecodealliance/wasmtime-rb/actions/runs/3662927895/jobs/6192358987), or [14 times as long](https://github.com/bytecodealliance/wasmtime-rb/actions/runs/3664528229/jobs/6195001928).

I don't have a Windows machine to try this out, but I _think_ the clock should still be fairly accurate there. I'm thinking (hoping) it might just be an issue with VMs / CI infrastructure.